how_to_edit_photo_image_programmaticaly.md

```diff
@@ -0,0 +1,166 @@
+keywords: [edit a photo, editing of pictures, edit an image, photo background editor, photo background change, resizing an image, resizing photos, photo sizing, picture resizer, image crop, crop photo, filter photo, online photo editor, online image editor, photo editing app, free photo editor]
+---
+
+# How to edit your photo or image programmatically
+
+## Intro
+
+In real life, we pay attention to colorful images, and in the virtual world, our perception inevitably reacts to a beautiful picture. The more attractive the shop window, the more likely we will go into the store itself; how the more interesting the visual content of the site, whether it is an online store or a thematic blog, the higher its traffic. However, the search and integration of engaging pictures is only part of the task facing the author, who forms the visual approach of the page. The next, and no less important step is editing photos and images. Properly edited photos - the same quality mark of the site as its text content. It increases website attractiveness, and user conversion rate, as a result creating new business opportunities for commercial sites and raising the recognition of researchers. The program library for photo editing is intended to facilitate the rapid development of new applications and services.
+
+Editing pictures is a frequent task for developers, researchers and students. Automatic editing of images helps to prepare initial raw photos for further processing. With programmable image processing, you can easily edit many photos at once. Tasks with photo editing are usually repeated and by using an image manipulation library you can significantly improve your performance. This process can be automatically done by using Aspose.Imaging program library. With the program library, it's easy to manipulate massively with a bunch of files. For example, repeat several operations sequences with images: image resizing, cropping, rotating and applying filters before publication on a website. Another popular case for programable image editing is creating a web service or custom standalone application for particular operations with images such are drawing text on images, removing the background or adjusting image parameters.
+
+In this article, we consider how to use the most demanded operations for image editing with Aspose.Imaging C# library:
+
+- [Resize](#resizing-an-image)
+- [Crop](#crop-an-image)
+- [Rotate](#rotate)
+- [Remove background](#remove-background)
+- [Filter](#filter-an-image)
+- [Merge](#merge-images)
+- [Grayscale](#grayscale-image)
+- [Binarize](#binarize-image)
+- [Dither](#dither-image)
+- [Adjust](#adjust-image)
+- [Cartoonify](#cartoonify-image)
+- [Drawing images](#drawing-images)
+- [Online editor](#how-to-edit-your-photo-or-image-online)
+
+Please follow the link to see the table with the list of all supported image formats -> [supported-file-formats](https://docs.aspose.com/imaging/net/supported-file-formats/)
```

Comment:
It is a markdown format for a hyperlink to a fragment on the same page:
[name](#fragment-link-ref)
It will be converted to html and not visible.

Reply:
@anms23 , As I see in html of your markdown https://aspose-imaging.github.io/aspose-imaging-probe/how_to_edit_photo_image_programmaticaly text link is with - is not it? Or what do you mean?

Reply:
@sam199008 , I mean if I omit the `#` mark, the markdown converter will process the link as a link to another file, not to a fragment reference of the same file. The hashtag is not visible on a page view, only in html source.

Reply:
@anms23 , I asked not properly maybe, I mean that format name1-name2-name3 with # looks like hashtag, i mean that maybe native format is  name1 name2 name3?

Reply:
@sam199008 , It will not work if I put spaces instead of hyphens. The link should be in a low case and spaces converted to hyphens:
`[Resize](#resizing-an-image)`
is the link to the heading:
`## Resizing an image`

Reply:
@anms23 , yes, it will be enought, it will be human readable text as expected.

Reply:
@anms23 , related to - usage in links I mean also all other links in document for example howto-crop-image etc

Reply:
@sam199008 , ok, fixed.
https://github.com/aspose-imaging/aspose-imaging-probe/pull/2/commits/72c0bb30f57a0fbd57e08a0fcf118227a22098c1

Reply:
@anms23 , Thanks, please note, when we will publish this article on production that need also separate howto to how to to be humanable text. Related to other questions, I do not have any additional questions, thank you, Imaging related probation you have succesfully passed.